packages/ckb-indexer/tests/collector.e2e.test.ts

```diff
@@ -32,6 +33,34 @@ test("get count correct", async (t) => {
   t.is(count, 1);
 });
 
+test("get count correct with multiple tests", async (t) => {
+  const type = {
+    codeHash:
+      "0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8",
+    hashType: "type" as HashType,
+    args: "0xa178db16d8228db82911fdb536df1916e761e205",
+  };
+  const cellCollector = new CellCollector(indexer, [
+    { lock: type },
+    {
+      lock: {
+        ...type,
+        args: "0x7ae354c3586ea3e7da6f30af80046fbe0cdce2fd",
+      },
+    },
+  ]);
+  const count = await cellCollector.count();
+  t.is(count, 2);
```

Comment:
```ts
const collected: Cell[] = []
for await (let cell of cellCollector.collect()) {
  collected.push(cell)
}

t.is(collected[0].lock.args, '0xa178db16d8228db82911fdb536df1916e761e205')
t.is(collected[1].lock.args, '0x7ae354c3586ea3e7da6f30af80046fbe0cdce2fd')
```

this is a less strict e2e case, the count of cells is 2 is possible that the same lock

Reply:
I think there are two other case for your concern.

https://github.com/ckb-js/lumos/pull/475/files#diff-61ebc5734333dcf70b862d371fe1b5f2ae917c40c1162d80a2706e9ca1fd017dR121

and

https://github.com/ckb-js/lumos/pull/475/files#diff-61ebc5734333dcf70b862d371fe1b5f2ae917c40c1162d80a2706e9ca1fd017dR96

Here, in this case I just test the count method, not the collect method. although the `count` method rely on `collect`.